### ISSUE TITLE
Diana - Precision loss can result in 0 output amount for calculateMint()

### ISSUE THREAD

**Issue by sherlock-admin** [sherlock-audit/2023-05-USSD-judging#468]
Diana

medium

# Precision loss can result in 0 output amount for calculateMint()

## Summary
Precision Loss while `calculateMint` can return 0 output

## Vulnerability Detail
Precision Loss while `calculateMint` can return 0 output.
In the expression: `(assetPrice * _amount) / 1e18)` if the Numerator < Denominator, then the output amount will be 0 and thus the user will be provided 0 STABLE COIN for provided _amount.

example: if `_amount = 1e18` and `assetPrice = 0.95`, their product will be less than the Denominator, and the output amount will be 0. So the user provides `_amount` tokens and in return receives 0 STABLE COIN

## Impact
The user provides `_amount` tokens and in return receives 0 STABLE COIN

## Code Snippet
https://github.com/sherlock-audit/2023-05-USSD/blob/main/ussd-contracts/contracts/USSD.sol#L172

```solidity
return (((assetPrice * _amount) / 1e18) * (10 ** decimals())) / (10 ** IERC20MetadataUpgradeable(_token).decimals());
```

## Tool used
Manual Review

## Recommendation
In the expression: `return (((assetPrice * _amount) / 1e18) * (10 ** decimals())) / (10 ** IERC20MetadataUpgradeable(_token).decimals());` , Multiplication should be done before division

Duplicate of #809
